statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,325 @@
+                                                  exposure=exposure,
+                                                  missing=missing, **kwargs)
+        self.exog = self.exog[self.endog >= (truncation + 1)]
+        self.endog = self.endog[self.endog >= (truncation + 1)]
+        self.trunc = truncation
+
+    def loglike(self, params):
+        """
+        Loglikelihood of Generic Truncated model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : float
+            The log-likelihood function of the model evaluated at `params`.
+            See notes.
+
+        Notes
+        --------
+        .. math:: \\ln L=\\sum_{y_{i}=0}\\ln(w_{i}+(1-w_{i})*P_{main\\_model})+
+            \\sum_{y_{i}>0}(\\ln(1-w_{i})+L_{main\\_model})
+            where P - pdf of main model, L - loglike function of main model.
+
+        """
+        return np.sum(self.loglikeobs(params))
+
+    def loglikeobs(self, params):
+        """
+        Loglikelihood for observations of Generic Truncated model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : ndarray (nobs,)
+            The log likelihood for each observation of the model evaluated
+            at `params`. See Notes
+
+        Notes
+        --------
+        .. math:: \\ln L=\\ln(w_{i}+(1-w_{i})*P_{main\\_model})+
+            \\ln(1-w_{i})+L_{main\\_model}
+            where P - pdf of main model, L - loglike function of main model.
+
+        for observations :math:`i=1,...,n`
+
+        """
+        llf_main = self.model_main.loglikeobs(params)
+
+        pmf = np.zeros_like(self.endog)
+        for i in range(self.trunc + 1):
+            model = self.model_main_name(np.ones_like(self.endog) * i,
+                                         self.exog)
+            pmf +=  np.exp(model.loglikeobs(params))
+
+        llf = llf_main - np.log(1 - pmf)
+
+        return llf
+
+    def score_obs(self, params):
+        """
+        Generic Truncated model score (gradient) vector of the log-likelihood
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+
+        Returns
+        -------
+        score : ndarray, 1-D
+            The score vector of the model, i.e. the first derivative of the
+            loglikelihood function, evaluated at `params`
+        """
+        score_main = self.model_main.score_obs(params)
+
+        pmf = np.zeros_like(self.endog)
+        score_trunc = np.zeros_like(score_main)
+        for i in range(self.trunc + 1):
+            model = self.model_main_name(np.ones_like(self.endog) * i,
+                                         self.exog)
+            pmf_i =  np.exp(model.loglikeobs(params))
+            score_trunc += (model.score_obs(params).T * pmf_i).T
+            pmf += pmf_i
```

Comment:
I'm not sure whether it will be very useful, but this could be put into a separate cdf_deriv method.

Overall, for the design: we should outsource parts to separate methods, if it is useful to override the method for specific models or distributions that have a analytical or computational more efficient way to calculate those parts.